### ISSUE TITLE
Threema Backup feature

### ISSUE THREAD

**Issue by simmac** [threema-ch/threema-web#65]
One of the things that make Threema quite difficult to handle for the average user is the backup procedure. You have to generate a full backup regularly on the phone, and then pull it from the phone on your PC. If you switch phones, you have to push the backup back to the new phone to import all messages and media.

It should be possible to implement a backup feature into the web client where you can generate a full Threema backup directly on your computer. This would make the whole backup procedure a whole lot less annoying.
An updated backup format could even enable incremental updates directly on the PC with automatic incremental backups every time the PC connects to the phone.

Especially the latter feature is complex and may require protocol updates, but I have no doubt that it will improve the usability of Threema a lot.

**Comment by rugk** [comment]
The issue is of course that the web client cannot save files directly without asking the user - for security reasons.
So automatic backups would be very difficult. You also don't really want to save these backups in local storage or so. They have a size limit and sitting storing these backups in your browser might have several other drawbacks…

However manual backups are a good idea. :+1:

**Comment by simmac** [comment]
Yes, the automatic/incremental updates would be a very-nice-to-have, but very difficult to implement if Threema Web will always be "browser-only".

**Comment by wschlich** [comment]
Is this still needed since Threema supports the native Android Backup service?!
Oh, well, I forgot iOS users...

**Comment by rugk** [comment]
Even for android users it's needed. Also in iOS most things are already backed up to iCloud AFAIK.
But not every android user wants to use the Google backup feature so this feature is really good.

**Comment by Buddabread** [comment]
Thsis would be very helpful in the way, that it's more comfortable for people to save their Backup.
On mobil devices every user have other possibilities to save the Backup (Mail, Bluetooth...). Would quide easy to tell my friends..first install Threema, make a Backup and than go into the webclient and save the Backup on the PC.
